### PR TITLE
domexception constructor reform upgrade

### PR DESCRIPTION
**Pull request by pmdartus**
This PR intregrates the changes made in https://github.com/jsdom/domexception/pull/3.

### CODE REVIEW

File: lib/jsdom/living/interfaces.js [hunk 1]
```diff
@@ -184,9 +185,15 @@ function install(window, name, interfaceConstructor) {
 }
 
 exports.installInterfaces = function (window) {
+  // TODO: Fix me
+  Object.defineProperty(window, "Error", {
```
Comment (pmdartus) [inline lib/jsdom/living/interfaces.js]: This line needs to be updated once we settle on an approach to expose ECMAScript globals (https://github.com/jsdom/jsdom/issues/2727). This is a tempory solution to make the tests pass.

**Comment by domenic** [comment]
Looks like one test is still failing, Document-createElementName-namespace.html

File: lib/jsdom/browser/js-globals.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
+[
```
Comment (domenic) [inline lib/jsdom/browser/js-globals.json]: Is this file supposed to be here? Should we wait on merging this until #2743 is merged?

Comment (pmdartus) [reply lib/jsdom/browser/js-globals.json]: I completely messed up the rebase... I will fix that tomorrow. But yes, let wait for your PR to get in first.

Comment (domenic) [reply lib/jsdom/browser/js-globals.json]: I'm ok landing in either order. If this lands first then #2743 just needs to delete a few extra lines, probably. I'm worried about getting stuck on #2743, although I've made some good progress I guess.

Comment (pmdartus) [reply lib/jsdom/browser/js-globals.json]: Let's wait for #2743 to land, you have made good progress there. I will try to help out on this, this week.

**Pull request merged**